tests/test_agents.py

```diff
@@ -1103,3 +1103,52 @@ async def test_continuation(self) -> None:
         # Check continuation of the search
         result = await tool.clinical_trials_search("Covid-19 vaccines", state)
         assert len(state.docs.docs) > trial_count, "Search was unable to continue"
+
+
+@pytest.mark.asyncio
+async def test_index_build_concurrency(agent_test_settings: Settings) -> None:
+    high_batch_size_save_index_count = 0
+
+    def high_batch_size_increment_save_index_count(*args, **kwargs):
+        nonlocal high_batch_size_save_index_count
+        high_batch_size_save_index_count += 1
+
+    high_concurrency_settings = agent_test_settings.model_copy(deep=True)
+    high_concurrency_settings.agent.index.name = "high_concurrency"
+    high_concurrency_settings.agent.index.concurrency = 3
+    high_concurrency_settings.agent.index.batch_size = 3
+    with patch.object(
+        SearchIndex,
+        "save_index",
+        side_effect=high_batch_size_increment_save_index_count,
+    ):
+        start_time = time.perf_counter()
+        await get_directory_index(settings=high_concurrency_settings)
+        high_concurrency_duration = time.perf_counter() - start_time
```

Comment:
```suggestion
    high_concurrency_settings = agent_test_settings.model_copy(deep=True)
    high_concurrency_settings.agent.index.name = "high_concurrency"
    high_concurrency_settings.agent.index.concurrency = 3
    high_concurrency_settings.agent.index.batch_size = 3
    with patch.object(
        SearchIndex,
        "save_index",
        side_effect=SearchIndex.save_index,
        autospec=True
    ) as mock_save_index:
        start_time = time.perf_counter()
        await get_directory_index(settings=high_concurrency_settings)
        high_concurrency_duration = time.perf_counter() - start_time
    high_batch_size_save_index_count = mock_save_index.awaited_count
```

The pass-through method is less code and more full stack (includes saving in the delta time)